cpp/min_cut/KargerMinCut.cpp

```diff
@@ -0,0 +1,98 @@
+#include <iostream>
+#include <vector>
+#include <cstdlib>
+#include <ctime>
+#include <climits>
+
+#include <min_cut/KargerMinCut.hpp>
+
+namespace Koala {
+
+KargerMinCut::KargerMinCut(int vertices, int repeat, const std::vector<std::vector<int>>& graphMatrix) : numVertices(vertices), numRepeat(repeat) {
+    srand(time(NULL));
+    
+    for (int i = 0; i < vertices; ++i) {
+        for (int j = 0; j < vertices; ++j) {
+            if (graphMatrix[i][j] > 0) {
+                addEdge(i, j, graphMatrix[i][j]);
+            }
+        }
+    }
+}
+
+void KargerMinCut::addEdge(int u, int v, int weight) {
+    for (int i = 0; i < weight; ++i) { // Adding multiple edges for the given weight
+        edges.push_back({u, v});
+    }
+}
+
+int KargerMinCut::find(std::vector<int>& parent, int i) {
+    if (parent[i] == i) return i;
+
+    return find(parent, parent[i]);
+}
+
+void KargerMinCut::unionSub(std::vector<int>& parent, std::vector<int>& rank, int x, int y) {
+    int xRoot = find(parent, x);
+    int yRoot = find(parent, y);
+
+    if (rank[xRoot] < rank[yRoot])
```

Comment:
Please reformat to stick to the [Egyptian Brackets](https://www.devx.com/terms/egyptian-brackets/) that is:
```
if (condition) {
  command;
}
```